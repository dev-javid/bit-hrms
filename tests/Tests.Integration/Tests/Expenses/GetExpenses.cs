namespace Tests.Integration.Tests.Expenses;

public class GetExpenses(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
{
    private const string Route = "api/expenses?page=1&limit=10";

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public async Task Should_return_expenses()
    {
        await LoginAsCompanyAdminAsync();
        await FeedDataAsync("Tests/Expenses/GetExpenses.sql");

        HttpResponseMessage response = await Client.GetAsync(Route);

        response.StatusCode.Should().Be(HttpStatusCode.OK);
        var result = await response.Content.ReadAsStringAsync();
        Approvals.VerifyJson(result);
    }
}
