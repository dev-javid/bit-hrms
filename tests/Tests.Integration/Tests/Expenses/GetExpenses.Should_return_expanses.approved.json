{
  "page": 1,
  "limit": 10,
  "total": 9,
  "items": [
    {
      "expenseId": 9,
      "companyId": 999,
      "amount": 1000,
      "documents": ["http://localhost/media/Expenses/9/test.png", "http://localhost/media/Expenses/9/test2.png"],
      "purpose": null,
      "createdOn": "2024-05-11T10:51:44.770234Z"
    },
    {
      "expenseId": 8,
      "companyId": 999,
      "amount": 1200,
      "documents": ["http://localhost/media/Expenses/8/test.png", "http://localhost/media/Expenses/8/test2.png"],
      "purpose": null,
      "createdOn": "2024-05-10T10:51:44.770234Z"
    },
    {
      "expenseId": 7,
      "companyId": 999,
      "amount": 1000,
      "documents": ["http://localhost/media/Expenses/7/test.png", "http://localhost/media/Expenses/7/test2.png"],
      "purpose": null,
      "createdOn": "2024-05-09T10:51:44.770234Z"
    },
    {
      "expenseId": 6,
      "companyId": 999,
      "amount": 1200,
      "documents": ["http://localhost/media/Expenses/6/test.png", "http://localhost/media/Expenses/6/test2.png"],
      "purpose": null,
      "createdOn": "2024-05-08T10:51:44.770234Z"
    },
    {
      "expenseId": 5,
      "companyId": 999,
      "amount": 1000,
      "documents": ["http://localhost/media/Expenses/5/test.png", "http://localhost/media/Expenses/5/test2.png"],
      "purpose": null,
      "createdOn": "2024-05-07T10:51:44.770234Z"
    },
    {
      "expenseId": 4,
      "companyId": 999,
      "amount": 1200,
      "documents": ["http://localhost/media/Expenses/4/test.png", "http://localhost/media/Expenses/4/test2.png"],
      "purpose": null,
      "createdOn": "2024-05-06T10:51:44.770234Z"
    },
    {
      "expenseId": 3,
      "companyId": 999,
      "amount": 1000,
      "documents": ["http://localhost/media/Expenses/3/test.png", "http://localhost/media/Expenses/3/test2.png"],
      "purpose": null,
      "createdOn": "2024-05-05T10:51:44.770234Z"
    },
    {
      "expenseId": 2,
      "companyId": 999,
      "amount": 1200,
      "documents": ["http://localhost/media/Expenses/2/test.png", "http://localhost/media/Expenses/2/test2.png"],
      "purpose": null,
      "createdOn": "2024-05-04T10:51:44.770234Z"
    },
    {
      "expenseId": 1,
      "companyId": 999,
      "amount": 1000,
      "documents": ["http://localhost/media/Expenses/1/test.png", "http://localhost/media/Expenses/1/test2.png"],
      "purpose": null,
      "createdOn": "2024-05-03T10:51:44.770234Z"
    }
  ]
}
